### ISSUE TITLE
tips for fixing [SettingWithCopyWarning] and suggestion for MACDH warning

### ISSUE THREAD

**Issue by zzwz02** [jealous/stockstats#62]
In `def _get_s`, change the last line of code (`StockDataFrame.set_nan(df[shifted_key], shift)`) to 
```
a=df[shifted_key].copy()
StockDataFrame.set_nan(a, shift)
df[shifted_key]=a.copy()
```
Same for `def _get_d`, change the last line of code (`StockDataFrame.set_nan(df[column_name], shift)`) to 
```
a=df[column_name].copy()
StockDataFrame.set_nan(a, shift)
df[column_name]=a.copy()
```
Should be the same for `def _get_p`, but I don't know which indicators are using this function, so I didn't test it. should change the last two lines of code:
```
StockDataFrame.set_nan(indices, shifts)
df[column_name] = indices
```

Also suggest to provide a flag to disable MACDH warning.

**Comment by jealous** [comment]
This is very useful.  Thank you very much!  I will put this into the next patch.

**Comment by jealous** [comment]
Fix in 0.3.2
